lib/turn.rb

```diff
@@ -0,0 +1,121 @@
+class Turn
+  attr_reader :player1, :player2, :spoils_of_war, :loser, :turn_winner
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+    @turn_winner = nil
+    @loser = loser
+  end
+  
+  # Return boolean value if either player has fewer than 3 cards left
+  
+  def fewer_than_3_cards?
+    if player1.deck.cards.count < 3 || player2.deck.cards.count < 3
+      true 
+    else
+      false
+    end
+  end
+  
+  # Return turn type based on the paired card rankings.
+  # Added :sudden_death type to account for scenario when a player does not have enough cards for a war.
+  
+  def type
+    if player1.deck.cards[0].rank == player2.deck.cards[0].rank && fewer_than_3_cards?
+      return :sudden_death
+    elsif player1.deck.cards[0].rank == player2.deck.cards[0].rank && player1.deck.cards[2].rank == player2.deck.cards[2].rank
+      return :mutually_assured_destruction
+    elsif player1.deck.cards[0].rank == player2.deck.cards[0].rank
+      return :war
+    else
+      return :basic
+    end
+  end
```

Comment:
I like the sequence of these conditionals, starting with checking if each player has enough cards to continue, and then moving onto the `:mutually_assured_destruction` scenario, and then `:war`, and finally `:basic`.